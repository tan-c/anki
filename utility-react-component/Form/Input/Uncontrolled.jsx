import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import { Map } from 'immutable';
import toastr from 'toastr';

export class InputUncontrolled extends React.Component {
  render() {
    const {
      inputName, record, actions, inputClassNames, pathName,
    } = this.props;
    return (
      <input
        type="text"
        name={inputName}
        className={inputClassNames}
        ref={(ref) => { this.ref = ref; }}
        placeholder={pathName.length ? record.getIn(pathName) : record.get(inputName)}
        onKeyDown={(event) => {
          if (event.which === 13) {
            const { value, name } = event.target;
            const newRecord = pathName.length ? record.setIn(pathName, value) : record.set(name, value);
            actions.update(newRecord).then((_) => {
              toastr.success(`${inputName} Updated successfully`);
              this.ref.value = '';
            });
          }
        }}
      />
    );
  }
}

InputUncontrolled.defaultProps = {
  pathName: [],
  inputClassNames: 'flex-1',
  inputName: '',
  record: Map(),
  actions: {},
};

InputUncontrolled.propTypes = {
  pathName: PropTypes.array,
  inputClassNames: PropTypes.string,
  inputName: PropTypes.string,
  record: PropTypes.object,
  actions: PropTypes.object,
};

function mapStateToProps(state, ownProps) {
  return {
    pathName: ownProps.pathName,
    inputName: ownProps.inputName,
    inputClassNames: ownProps.inputClassNames,
    record: ownProps.record,
    actions: ownProps.actions,
  };
}

// function mapDispatchToProps(dispatch) {
//   return {
//   };
// }

export default connect(mapStateToProps)(InputUncontrolled);
