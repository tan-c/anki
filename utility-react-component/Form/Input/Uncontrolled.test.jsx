import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { InputUncontrolled } from './Uncontrolled';

function setup() {
  const props = {
  };

  return shallow(<InputUncontrolled {...props} />);
}

describe('InputUncontrolled Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find('input')).toHaveLength(1);
    });
  });
});
