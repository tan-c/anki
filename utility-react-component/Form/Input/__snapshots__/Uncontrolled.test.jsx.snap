// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputUncontrolled Unit Tests when normally set up renders correctly 1`] = `
<input
  className="flex-1"
  name=""
  onKeyDown={[Function]}
  type="text"
/>
`;
